lib/services/network/http_service.dart

```diff
@@ -1,14 +1,224 @@
 import 'package:bluebubbles/models/models.dart';
+import 'package:bluebubbles/utils/file_utils.dart';
 import 'package:bluebubbles/utils/logger.dart';
 import 'package:bluebubbles/services/services.dart';
 import 'package:dio/dio.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:get/get.dart' hide Response, FormData, MultipartFile;
+import 'backend_service.dart';
 
 /// Get an instance of our [HttpService]
 HttpService http = Get.isRegistered<HttpService>() ? Get.find<HttpService>() : Get.put(HttpService());
 
+class HttpBackend implements BackendService {
```

Comment:
Couple of things for this class:

1. Would it make sense to have this as a separate file
2. This needs proper error handling in any http call. If an HTTP call completes with something not code 200, it will throw a `Future.error`. Currently your code does not account for that. To account for this, wrap your `http.xxx` call in `try/catch` and in the catch make sure you also return `Future.error` so the error is passed downstream to whichever file ran the async function.